### PR TITLE
[home] implement redesigned project details and branch-related screens

### PR DESCRIPTION
**Pull request by FiberJW**
# Why

The new design includes updates to the project details

# How

Almost all of the logic is a duplication of the previous project page. This PR mainly splits out the duplicated Project Screen file into one file per component and updates the UI components with DCC to match the new designs.

# Test Plan

Open the project page by clicking on of the 'projects' on the home page, not the 'recently' opened projects because that will attempt to launch it instead of opening the details page:

IMG_4800

IMG_4799

IMG_4798

IMG_4797

Screenshot_20220319-205229

### CODE REVIEW

File: home/components/BranchListItem.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+ */
+
+export function BranchListItem({
+  name,
+  latestUpdate: { message, manifestPermalink, createdAt },
+}: Props) {
+  const theme = useExpoTheme();
+
+  const handlePressBranch = () => {
+    // TODO: open branch page
```
Comment (ajsmth) [inline home/components/BranchListItem.tsx:32]: is this done by the invocation below?

File: home/constants/DateFormats.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+// https://date-fns.org/v2.11.0/docs/format
```
Comment (ajsmth) [inline home/constants/DateFormats.ts:1]: love this idea - I will have to add a formats obj in all my projects in the future

File: home/screens/RedesignedProjectScreen/WarningBox.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        onPress={onLearnMorePress}
+        containerProps={{
+          style: {
+            padding: spacing[2],
+            alignSelf: 'flex-start',
+            backgroundColor: theme.button.tertiary.background,
+          },
+          rounded: 'small',
+        }}>
+        <Text type="InterSemiBold" style={{ color: theme.button.tertiary.foreground }} size="small">
```
Comment (ajsmth) [inline home/screens/RedesignedProjectScreen/WarningBox.tsx:35]: I think you can also do `<Button.Text color="tertiary" />` if you wanted

**Review by ajsmth — APPROVED** [review]
this looks great!

File: home/components/BranchListItem.tsx (hunk 1 above)
Comment (ajsmth) [reply home/components/BranchListItem.tsx:32]: disregard - I see you updated this in already in a separate PR

File: home/screens/RedesignedProjectScreen/WarningBox.tsx (hunk 3 above)
Comment (FiberJW) [reply home/screens/RedesignedProjectScreen/WarningBox.tsx:35]: I think I was having problems with Button.text which is why I did this

Comment (FiberJW) [reply home/screens/RedesignedProjectScreen/WarningBox.tsx:35]: It was showing the wrong color in dark mode

**Pull request merged**